client/chat/chat-reducer.ts

```diff
@@ -33,33 +34,39 @@ export interface MessagesState {
 }
 
 export interface ChatState {
-  /* A set of joined chat channels */
+  /** A set of joined chat channels */
   joinedChannels: Set<SbChannelId>
-  /* A map of channel ID -> channel info (includes both joined and unjoined chat channels) */
+  /** A map of channel ID -> channel info (includes both joined and unjoined chat channels) */
   idToInfo: Map<SbChannelId, ChannelInfo>
-  /* A map of channel ID -> channel users */
+  /** A map of channel ID -> joined channel data (only has data for user's joined channels) */
+  idToJoinedData: Map<SbChannelId, JoinedChannelData>
+  /** A map of channel ID -> channel users */
   idToUsers: Map<SbChannelId, UsersState>
-  /* A map of channel ID -> channel messages */
+  /** A map of channel ID -> channel messages */
   idToMessages: Map<SbChannelId, MessagesState>
-  /* A nested map of channel ID -> a map of user ID -> chat channel user profile */
+  /** A nested map of channel ID -> a map of user ID -> chat channel user profile */
   idToUserProfiles: Map<SbChannelId, Map<SbUserId, ChatUserProfileJson>>
-  /* A map of channel ID -> your own permissions for this chat channel */
+  /** A map of channel ID -> your own permissions for this chat channel */
   idToSelfPermissions: Map<SbChannelId, ChannelPermissions>
-  /* A set of joined chat channels that are activated */
+  /** A set of joined chat channels that are activated */
   activatedChannels: Set<SbChannelId>
-  /* A set of joined chat channels that are unread */
+  /** A set of joined chat channels that are unread */
   unreadChannels: Set<SbChannelId>
+  /** A set of channel IDs saved in various chat messages that no longer exist. */
+  deletedChannels: Set<SbChannelId>
 }
 
 const DEFAULT_CHAT_STATE: Immutable<ChatState> = {
   joinedChannels: new Set(),
   idToInfo: new Map(),
+  idToJoinedData: new Map(),
```

Comment:
Not an issue for this actual line, but there's an event for leaving channels, I think? You need to use that to delete the data for that channel from this map